core/ecschema-editing/src/Editing/RelationshipClasses.ts

```diff
@@ -107,15 +108,18 @@ export class RelationshipClasses extends ECClasses {
    * @param baseClassKey The SchemaItemKey of the base class. Specifying 'undefined' removes the base class.
    */
   public override async setBaseClass(itemKey: SchemaItemKey, baseClassKey?: SchemaItemKey): Promise<void> {
-    const relClass = await this.schemaEditor.schemaContext.getSchemaItem<RelationshipClass>(itemKey);
+    const relClass = await this.schemaEditor.schemaContext.getSchemaItem<RelationshipClass>(itemKey)
+      .catch((e) => {
+        throw new SchemaEditingError(ECEditingStatus.SetBaseClass, new ClassId(this.schemaItemType, itemKey), e);
+      });
     const baseClass = relClass?.baseClass;
 
     await super.setBaseClass(itemKey, baseClassKey);
 
     try {
       await this.validate(relClass!);
     } catch(e: any) {
-      await relClass!.setBaseClass(baseClass);
+      await (relClass! as ECClass as MutableClass).setBaseClass(baseClass);
```

Comment:
why is a double cast necessary? especially given you already have  the definitely assigned operator, which together makes this about the least typescript typescript way to accomplish what you seem to want to do here